src/Parser/Hunter/Marksmanship/Modules/Spells/Trueshot.js

```diff
@@ -23,23 +24,46 @@ class Trueshot extends Analyzer {
   totalCritsInTS = 0;
   totalCastsPrTS = 0;
   executeTrueshots = 0;
+  startFocusForCombatant = 0;
+  prepullTrueshots = 0;
 
-  on_byPlayer_cast(event) {
+  on_byPlayer_applybuff(event) {
     const buffId = event.ability.guid;
-    if (buffId !== SPELLS.TRUESHOT.id) {
+    if (buffId !== SPELLS.TRUESHOT.id && !event.prepull) {
```

Comment:
Shouldn't this be `||` not, `&&` ?

Reply:
Well I only care about prepull events and trueshot events in this section, since everything else is handled in ``on_byPlayer_cast``

Reply:
Yes, you only care about an event that is Trueshot AND Prepull, but this statement is for an early return. The logical negation of "A && B" is "!A || !B". For a concrete example, a buff that is Prepull but isn't Trueshot would not early return as currently written.

Reply:
right... fuck me - fixed